File: CS/ImageTextEdit/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace ImageTextEdit
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void customIconTextEdit1_Properties_OnIconSelection(object sender, OnIconSelectionEventArgs e)
        {
            CustomIconTextEdit editor = sender as CustomIconTextEdit;
            if (editor != null)
            {
                if (editor.Enabled)
                    if (editor.ContainsFocus)
                        e.ImageIndex = 0;
                    else
                        e.ImageIndex = 1;
                else
                    e.ImageIndex = 2;

            }
        }

        private void customIconTextEdit1_IconClick(object sender, MouseEventArgs e)
        {
            CustomIconTextEdit editor = sender as CustomIconTextEdit;
            if (editor != null)
            {
                this.Text = editor.Name + ".Text = " + editor.Text;
            }
        }
    }
}

File: VB/ImageTextEdit/Form1.vb
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms

Namespace ImageTextEdit

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub customIconTextEdit1_Properties_OnIconSelection(ByVal sender As Object, ByVal e As OnIconSelectionEventArgs)
            Dim editor As CustomIconTextEdit = TryCast(sender, CustomIconTextEdit)
            If editor IsNot Nothing Then
                If editor.Enabled Then
                    If editor.ContainsFocus Then
                        e.ImageIndex = 0
                    Else
                        e.ImageIndex = 1
                    End If
                Else
                    e.ImageIndex = 2
                End If
            End If
        End Sub

        Private Sub customIconTextEdit1_IconClick(ByVal sender As Object, ByVal e As MouseEventArgs)
            Dim editor As CustomIconTextEdit = TryCast(sender, CustomIconTextEdit)
            If editor IsNot Nothing Then
                Text = editor.Name & ".Text = " & editor.Text
            End If
        End Sub
    End Class
End Namespace
